includes/Core/Util/Entity_Factory.php

```diff
@@ -484,4 +497,49 @@ private static function prefix_title( $title, $prefix ) {
 			$title
 		);
 	}
+
+	/**
+	 * Gets the entity for the given URL, if available.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $url URL to determine the entity from.
+	 * @param Entity $entity The initial entity.
+	 * @return Entity The initial or new entity for the given URL.
+	 */
+	private static function maybe_convert_to_amp_entity( $url, $entity ) {
+		$url_parts    = wp_parse_url( $url );
+		$new_url_tail = '';
+
+		wp_parse_str( $url_parts['query'], $url_query_params );
```

Comment:
Let's define the `$url_query_params` variable before passing it to the function, plus we need to call the function only if the `query` property exists in the array:
```suggestion
		$url_query_params = array();
		if ( ! empty( $url_parts['query'] ) {
			wp_parse_str( $url_parts['query'], $url_query_params );
		}
```